exchange/bidder.go

```diff
@@ -169,11 +171,11 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, bidderRequest Bidde
 		dataLen = len(reqData) + len(bidderRequest.BidderStoredResponses)
 		responseChannel = make(chan *httpCallInfo, dataLen)
 		if len(reqData) == 1 {
-			responseChannel <- bidder.doRequest(ctx, reqData[0])
+			responseChannel <- bidder.doRequest(ctx, reqData[0], reqInfo.BidderRequestStartTime, metrics.NonErrorPreBidderRequest)
```

Comment:
I'm not sure if are we planning to record some other type of overhead in the future. Should we keep it simple for now and remove the third parameter from `bidder.doRequest`? We could simply log the `RecordOverheadTime` metric with the `metrics.NonErrorPreBidderRequest` overhead as a constant.
```
173         if len(reqData) == 1 {
174 -           responseChannel <- bidder.doRequest(ctx, reqData[0], reqInfo.BidderRequestStartTime, metrics.NonErrorPreBidderRequest)
    +           responseChannel <- bidder.doRequest(ctx, reqData[0], reqInfo.BidderRequestStartTime)
175         } else {
176             for _, oneReqData := range reqData {
177                 go func(data *adapters.RequestData) {
178 -                   responseChannel <- bidder.doRequest(ctx, data, reqInfo.BidderRequestStartTime, metrics.NonErrorPreBidderRequest)
    +                   responseChannel <- bidder.doRequest(ctx, data, reqInfo.BidderRequestStartTime)
179                 }(oneReqData) // Method arg avoids a race condition on oneReqData
180             }
181         }
                  .
                  .
490
491   // doRequest makes a request, handles the response, and returns the data needed by the
492   // Bidder interface.
493 - func (bidder *bidderAdapter) doRequest(ctx context.Context, req *adapters.RequestData, pbsRequestStartTime time.Time, overhead metrics.OverheadType) *httpCallInfo {
    + func (bidder *bidderAdapter) doRequest(ctx context.Context, req *adapters.RequestData, pbsRequestStartTime time.Time) *httpCallInfo {
494 -     return bidder.doRequestImpl(ctx, req, glog.Warningf, pbsRequestStartTime, overhead)
    +     return bidder.doRequestImpl(ctx, req, glog.Warningf, pbsRequestStartTime)
495   }
496
497   func (bidder *bidderAdapter) doRequestImpl(ctx context.Context, req *adapters.RequestData, logger util.LogMsg, pbsRequestStartTime time.Time) *httpCallInfo {
498       var requestBody []byte
                  .
                  .
518       if !bidder.config.DisableConnMetrics {
519           ctx = bidder.addClientTrace(ctx)
520       }
521 -     bidder.me.RecordOverheadTime(overhead, time.Since(pbsRequestStartTime))
    +     bidder.me.RecordOverheadTime(metrics.NonErrorPreBidderRequest, time.Since(pbsRequestStartTime))
522       httpResp, err := ctxhttp.Do(ctx, bidder.Client, httpReq)

exchange/bidder.go
```

Reply:
Agree, good point

Reply:
https://github.com/prebid/prebid-server/pull/2658/commits/3a0bcf0d50ff710b0df46154687ca9e9bfdcae51 makes the specified changes. thanks